Mlem/Views/Tabs/Profile/UserView+Logic.swift

```diff
@@ -10,6 +10,22 @@ import SwiftUI
 extension UserView {
     var isOwnProfile: Bool { user.userId == siteInformation.myUserInfo?.localUserView.person.id }
     
+    var menuFunctions: [MenuFunction] {
+        var ret = user.menuFunctions({ user = $0 }, modToolTracker: modToolTracker)
+        
+        // add moderator needs to be defined as a menu function out here instead of within the user model so that it can take the $user binding
+        // TODO: 2.0 move add moderator menu function into Person
+        if !isOwnProfile || siteInformation.myUser?.isAdmin ?? false {
+            ret.append(.standardMenuFunction(
+                text: "Appoint as Moderator",
+                imageName: Icons.moderation
+            ) {
```

Comment:
We should hide this menu function if the user does not moderate any communities and is not an admin

Reply:
Good call, implemented 👍 

I've also added a waiting overlay and cancel buttons to the search view